Introduce PTEFile class

lucylq opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #15800

PTEFile class holds the components of a PTE file: the program, mutable constants and named data.

Currently, the `program` definition does not contain mutable constants and named data; they are always stored in segments and not inline. This means when we deserialize, they are lost, because we only deserialize into the `program` concept.

Now, segment data is included in the PTEFile class.

Differential Revision: [D86814175](https://our.internmc.facebook.com/intern/diff/D86814175/)

lucylq committed:
Introduce PTEFile class

PTEFile class holds the components of a PTE file: the program, mutable constants and named data.

Currently, the `program` definition does not contain mutable constants and named data; they are always stored in segments and not inline. This means when we deserialize, they are lost, because we only deserialize into the `program` concept.

Now, segment data is included in the PTEFile class.

Differential Revision: [D86814175](https://our.internmc.facebook.com/intern/diff/D86814175/)

[ghstack-poisoned]

lucylq committed:
Update on "Introduce PTEFile class"


PTEFile class holds the components of a PTE file: the program, mutable constants and named data.

Currently, the `program` definition does not contain mutable constants and named data; they are always stored in segments and not inline. This means when we deserialize, they are lost, because we only deserialize into the `program` concept.

Now, segment data is included in the PTEFile class.

Differential Revision: [D86814175](https://our.internmc.facebook.com/intern/diff/D86814175/)

[ghstack-poisoned]

zingo reviewed (commented):
It might be nice it bundleIO BPTE files could be mixed into this in some way and handled similarly.

lucylq commented:
> It might be nice it bundleIO BPTE files could be mixed into this in some way and handled similarly.

@zingo thanks for the suggestion - could you comment a bit more on what you're looking for?

zingo commented:
> > It might be nice it bundleIO BPTE files could be mixed into this in some way and handled similarly.
> 
> @zingo thanks for the suggestion - could you comment a bit more on what you're looking for?

Im thinking the it might be god to have if you made tools parsing the different files. As Im not writing those kind of tools right now Im not sure its fully applicable but I feel the bpte are sometimes forgotten or need special handling.

lucylq committed:
Update on "Introduce PTEFile class"


PTEFile class holds the components of a PTE file: the program, mutable constants and named data.

Currently, the `program` definition does not contain mutable constants and named data; they are always stored in segments and not inline. This means when we deserialize, they are lost, because we only deserialize into the `program` concept.

Now, segment data is included in the PTEFile class.

Differential Revision: [D86814175](https://our.internmc.facebook.com/intern/diff/D86814175/)

[ghstack-poisoned]

lucylq committed:
Update on "Introduce PTEFile class"


PTEFile class holds the components of a PTE file: the program, mutable constants and named data.

Currently, the `program` definition does not contain mutable constants and named data; they are always stored in segments and not inline. This means when we deserialize, they are lost, because we only deserialize into the `program` concept.

Now, segment data is included in the PTEFile class.

Differential Revision: [D86814175](https://our.internmc.facebook.com/intern/diff/D86814175/)

[ghstack-poisoned]

lucylq committed:
Update on "Introduce PTEFile class"


PTEFile class holds the components of a PTE file: the program, mutable constants and named data.

Currently, the `program` definition does not contain mutable constants and named data; they are always stored in segments and not inline. This means when we deserialize, they are lost, because we only deserialize into the `program` concept.

Now, segment data is included in the PTEFile class.

Differential Revision: [D86814175](https://our.internmc.facebook.com/intern/diff/D86814175/)

[ghstack-poisoned]

lucylq committed:
Update on "Introduce PTEFile class"


PTEFile class holds the components of a PTE file: the program, mutable constants and named data.

Currently, the `program` definition does not contain mutable constants and named data; they are always stored in segments and not inline. This means when we deserialize, they are lost, because we only deserialize into the `program` concept.

Now, segment data is included in the PTEFile class.

Differential Revision: [D86814175](https://our.internmc.facebook.com/intern/diff/D86814175/)

[ghstack-poisoned]

lucylq committed:
Update on "Introduce PTEFile class"


PTEFile class holds the components of a PTE file: the program, mutable constants and named data.

Currently, the `program` definition does not contain mutable constants and named data; they are always stored in segments and not inline. This means when we deserialize, they are lost, because we only deserialize into the `program` concept.

Now, segment data is included in the PTEFile class.

Differential Revision: [D86814175](https://our.internmc.facebook.com/intern/diff/D86814175/)

[ghstack-poisoned]

lucylq committed:
Update on "Introduce PTEFile class"


PTEFile class holds the components of a PTE file: the program, mutable constants and named data.

Currently, the `program` definition does not contain mutable constants and named data; they are always stored in segments and not inline. This means when we deserialize, they are lost, because we only deserialize into the `program` concept.

Now, segment data is included in the PTEFile class.

Differential Revision: [D86814175](https://our.internmc.facebook.com/intern/diff/D86814175/)

[ghstack-poisoned]

lucylq committed:
Update on "Introduce PTEFile class"


PTEFile class holds the components of a PTE file: the program, mutable constants and named data.

Currently, the `program` definition does not contain mutable constants and named data; they are always stored in segments and not inline. This means when we deserialize, they are lost, because we only deserialize into the `program` concept.

Now, segment data is included in the PTEFile class.

Differential Revision: [D86814175](https://our.internmc.facebook.com/intern/diff/D86814175/)

[ghstack-poisoned]
